packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/repository.ts

```diff
@@ -762,6 +773,285 @@ export class SavedObjectsRepository implements ISavedObjectsRepository {
     );
   }
 
+  /**
+   * Performs initial checks on object type validity and flags multi-namespace objects for preflight checks by adding an `esRequestIndex`
+   * @param objects SavedObjectsBulkDeleteObject[]
+   * @returns array BulkDeleteExpectedBulkGetResult[]
+   * @internal
+   */
+  private presortObjectsByNamespaceType(objects: SavedObjectsBulkDeleteObject[]) {
```

Comment:
I extracted this from the main body of the `bulkDelete` implementation to make it easier to follow. The intent is to eventually extract all the pre-flight handling to another module, similar to what's been done for `bulkResolve`.